thentos-core/src/Servant/Missing.hs

```diff
@@ -41,6 +42,12 @@ data FormReqBody
 fromEnvIdentity :: Applicative m => Env Identity -> Env m
 fromEnvIdentity env = pure . runIdentity . env
 
+-- | FIXME: not sure this is legal.  do we need to make the IORef thread safe?  by wrapping it into
+-- an STRef?  but it seems like we are using it exactly like it is intended to.
```

Comment:
The single, unprotected IORef seems correct --- looking at implementation of resourceForkIO, we see only a single IORef is passed to all the resourceForkIO-spawned threads and since it's a single IORef, the atomicModifyIORef wrappers, that it's protected with, are enough to prevent race conditions, etc. The question remains whether we should use/reimplement resourceForkIO. Investigating...